### DISCUSSION TITLE
Authentication Hell

### DISCUSSION THREAD

**Discussion by ghost** [community/community#84742] [New to GitHub] [answered]
Hi,

So I haven't developed code in like 10 years.  So I'm trying to `git push origin main` and it asks for my username and password.  Apparently it doesn't accept my username and password, because github no longer authenticates that way.

1. Why is it asking for my password if that's not good enough?

The link it provides when I have this issue sends me to github's general help for cloning a repo, which provides absolutely nothing on how to cope with the authentication problem.

2. Why is it sending me to https://docs.github.com/en/get-started/getting-started-with-git/about-remote-repositories#cloning-with-https-urls when this URL itself has literally nothing on setting up authentication to allow a `git push origin master`?  Instead it links off to dozens of other documents forming something like the head of the Canadian Tax Code.

From reading over 100 pages of documentation, I have learned that I somehow need to set up SSH and then somehow bind that to github.  To upload a hobby project to a folder.  Somehow.  Still no word on how, but I'm expecting it to be another 100 pages of sifting through documentation to figure out.

3. Why is it so difficult to authenticate to set up a hobby project on github?  My banking, tax ID number and credit history combined have less difficulty authenticating to than github?  Why do I need to spend an entire 8 hour day figuring out how to authenticate?  Why can't it be obvious?  

4. Is there some hidden page somewhere nested deep in my Settings that allows me to set up the things I need step-by-step so that I can use my own repo?

**Comment by ghost** [comment, marked as answer]
To create an authentication token, you go to Settings here on the right:
![image](https://github.com/community/community/assets/12051895/a466fcf4-6bfe-4077-a351-3e6d5dd730f8)

Then to Personal Access Tokens here on the left:
![image](https://github.com/community/community/assets/12051895/f38ef2f4-95da-49a5-be76-de9e7905ece6)

Then you want to create a Classic Token.  I leave you on your own to figure that.

Once you have one, save the token somewhere along with your passwords.

Then go to your command line.

You should already have your git name and email set up.

`git init` was already figured, and essentially starts git managing a folder.  You can then `git pull` into it rather readily without authentication issues.

If you add any files to your local folder you were wanting to add to to a repo, you would use `git add .`

This was the one I couldn't find:

`git remote set-url origin https://[insert account name here without square brackets]:[insert token here without square brackets]@github.com/[insert account name here without square brackets]/[insert project name here without square brackets]`

I would replace those with real values, but it's authentication, you'll have to make do from there.  That was the magic line I seemed to have needed to be able to authenticate. 

Then I needed to do a `git commit -m "message"`.  This basically tells git there's an update in the local repository, and if you push without it, it won't know what to do.

Then to push I can use `git push -u origin main`.  -u is to update, and it should basically only let you push if the local commit doesn't conflict with what's on the site (it shouldn't unless someone else has updated it since)


If you need help, feel free to ask me, or you can also ask ChatGPT, as it seems to provide more relevant answers than the documentation.

**Comment by RiturajAzad-RA** [comment]
after applying all above steps, I'm still getting below error
"**error: RPC failed; HTTP 403 curl 22 The requested URL returned error: 403
send-pack: unexpected disconnect while reading sideband packet
fatal: the remote end hung up unexpectedly**"

git commit is success for me, git push gives the above error
